working-with-collections/maps.md

```diff
@@ -1,9 +1,9 @@
 # ![ears](../img/ears.png) Maps
 
-Just like Hop provides an immutable slice type and various methods for creating
-new slices from existing ones, it also provides a map collection. Like slices,
-maps can store any type of value. However, they only allow either `int` or
-`string` keys.
+Just like stores provide an immutable slice type and various methods for
+creating new slices from existing ones, it also provides a map collection. Like
+slices, maps can store any type of value. However, they only allow either `int`
+or `string` keys.
```

Comment:
I see that we did in fact limit the key to `int` and `string` [here](https://github.com/nextmv-io/sdk/blob/77b609861f250fa187128c824184e7304ce511d0/store/map.go#L8). Is there any reason why we wouldn't support `go`'s own `comparable` [type](https://go.dev/ref/spec#Predeclared_identifiers)?

Reply:
Opening up a ticket to track this. I do remember this was a design choice by Ryan.

Reply:
It was a response to a limitation, not a design choice. Commenting in ticket.